### PR TITLE
[cxx-interop] Allow more C++ decls in public Swift interfaces 

### PR DESCRIPTION
**Pull request by egorzhdan**
This is a follow-up to 8859b629.

Let's allow usages of C++ global variables, fields and constructors, as long as the types involved aren't fragile, e.g. if they are C-like structs.

This resolves compiler errors when trying to rebuild System.swiftmodule from its textual interface with Xcode 16.1.

rdar://140203932
rdar://141124318

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please test

File: lib/Sema/TypeCheckAccess.cpp [hunk 1]
```diff
@@ -1891,9 +1894,21 @@ bool isFragileClangNode(const ClangNode &node) {
     return !cxxRecordDecl->isCLike() &&
            !cxxRecordDecl->getDeclContext()->isExternCContext();
   }
+  if (auto *varDecl = dyn_cast<clang::VarDecl>(decl))
+    return isFragileClangType(varDecl->getType());
+  if (auto *fieldDecl = dyn_cast<clang::FieldDecl>(decl))
+    return isFragileClangType(fieldDecl->getType()) &&
```
Comment (tshortli) [inline lib/Sema/TypeCheckAccess.cpp:1900]: Should this be `||` instead of `&&`?

Comment (egorzhdan) [reply lib/Sema/TypeCheckAccess.cpp]: Oh, right, thanks for catching this!

**Review by tshortli — APPROVED** [review]

**Comment by egorzhdan** [comment]
@swift-ci please test

**Comment by egorzhdan** [comment]
@swift-ci please test Windows

**Review by Xazax-hun — APPROVED** [review]
LGTM!

**Pull request merged**